lib/routes/utils/request_helper.js

```diff
@@ -9,9 +9,32 @@
  * @returns {boolean}
  */
 function wantsKeys (request) {
-  return request.query.keys === 'true'
+  return request.query.keys === 'true' || request.query.keys === true
+}
+
+/**
+ * Returns whether or not to perform a signin verification email.
+ * Currently, the uid is converted to a number between 0-100 which corresponds to the
+ * sign-in confirmation sample rate.
+ *
+ * @param uid
+ * @returns {boolean}
+ */
+function shouldEnableSigninConfirmation(uid, config) {
+  // Sign-in confirmation is determined by whether the feature is enabled and
+  // if the user's first two uid digits fall within the config sample rate.
+  if (config.signinConfirmation && config.signinConfirmation.enabled) {
+
+    // Returns the first 2 integers of uid
+    var uidNum = parseInt(uid.substr(0, 4), 16) % 100
```

Comment:
Cool, thanks. To properly satisfy myself, I tried it in the repl. Looks pretty uniform to me!

```
> const distribution = new Array(100).fill(0)
undefined
> const uuid = require('uuid')
undefined
> for (var i = 0; i < 1000000; ++i) {
... ++distribution[parseInt(uuid.v4('binary').toString('hex').substr(0, 4), 16) % 100]
... }
9917
[ 10007,
  10078,
  9856,
  9901,
  10072,
  10006,
  9916,
  9832,
  10006,
  10190,
  10103,
  9941,
  10034,
  9811,
  10026,
  10084,
  9804,
  9911,
  10031,
  9927,
  9860,
  10069,
  9857,
  9722,
  10082,
  9962,
  10054,
  10125,
  10114,
  10081,
  10076,
  10024,
  9946,
  10029,
  10002,
  10193,
  9943,
  10100,
  9992,
  9890,
  10027,
  9883,
  9991,
  10165,
  9929,
  10013,
  9948,
  10138,
  9966,
  9985,
  10146,
  9745,
  10156,
  10020,
  9881,
  9859,
  10017,
  10068,
  10124,
  10045,
  9905,
  10134,
  10038,
  10098,
  10051,
  10073,
  10134,
  10064,
  10086,
  9972,
  9914,
  9830,
  9923,
  9917,
  10003,
  10047,
  9837,
  9950,
  9945,
  10017,
  9859,
  10100,
  9944,
  10171,
  10132,
  10108,
  9946,
  10034,
  9917,
  9999,
  9972,
  9828,
  9992,
  9992,
  10051,
  10126,
  9970,
  10130,
  10073,
  9955 ]
```